### ISSUE TITLE
Internal error: encountered unexpected error in runtime on Julia latest master

### ISSUE THREAD

**Issue by Gnimuc** [quinnj/JSON2.jl#8]
This is actually not an error, everything works just fine except those awful messages.

```julia
julia> versioninfo()
Julia Version 0.7.0-alpha.59
Commit 2a45839ef9 (2018-06-09 17:04 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin17.6.0)
  CPU: Intel(R) Core(TM) i5-6267U CPU @ 2.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = subl
  JULIA_NUM_THREADS = 4
```
```julia
julia> struct T
           a::Int
           b::Int
           c::Union{Nothing, Int}
       end

julia> JSON2.read("""{"a": 0, "b": 1, "c": null, "d": 3}""")
Successor 140 of block 139 not in predecessor list
Internal error: encountered unexpected error in runtime:
MethodError(f=typeof(Base.string)(), args=(), world=0x0000000000000f26)
rec_backtrace at /Users/gnimuc/Codes/julia/src/stackwalk.c:94
record_backtrace at /Users/gnimuc/Codes/julia/src/task.c:246 [inlined]
jl_throw at /Users/gnimuc/Codes/julia/src/task.c:577
jl_method_error_bare at /Users/gnimuc/Codes/julia/src/gf.c:1608
jl_method_error at /Users/gnimuc/Codes/julia/src/gf.c:1626
jl_lookup_generic_ at /Users/gnimuc/Codes/julia/src/gf.c:2117
jl_apply_generic at /Users/gnimuc/Codes/julia/src/gf.c:2137
error at ./error.jl:42
verify_ir at ./compiler/ssair/verify.jl:67
run_passes at ./compiler/ssair/driver.jl:140
optimize at ./compiler/optimize.jl:172
typeinf at ./compiler/typeinfer.jl:451
typeinf_edge at ./compiler/typeinfer.jl:137
abstract_call_method at ./compiler/abstractinterpretation.jl:303
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:78
abstract_call at ./compiler/abstractinterpretation.jl:710
abstract_eval_call at ./compiler/abstractinterpretation.jl:739
abstract_eval at ./compiler/abstractinterpretation.jl:811
typeinf_work at ./compiler/typeinfer.jl:361
typeinf at ./compiler/typeinfer.jl:422
typeinf_edge at ./compiler/typeinfer.jl:137
abstract_call_method at ./compiler/abstractinterpretation.jl:303
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:78
abstract_call at ./compiler/abstractinterpretation.jl:710
abstract_apply at ./compiler/abstractinterpretation.jl:443
abstract_call at ./compiler/abstractinterpretation.jl:492
abstract_eval_call at ./compiler/abstractinterpretation.jl:739
abstract_eval at ./compiler/abstractinterpretation.jl:811
typeinf_work at ./compiler/typeinfer.jl:375
typeinf at ./compiler/typeinfer.jl:422
typeinf_edge at ./compiler/typeinfer.jl:137
abstract_call_method at ./compiler/abstractinterpretation.jl:303
abstract_call_gf_by_type at ./compiler/abstractinterpretation.jl:78
abstract_call at ./compiler/abstractinterpretation.jl:710
abstract_eval_call at ./compiler/abstractinterpretation.jl:739
abstract_eval at ./compiler/abstractinterpretation.jl:811
typeinf_work at ./compiler/typeinfer.jl:375
typeinf at ./compiler/typeinfer.jl:422
typeinf_frame at ./compiler/typeinfer.jl:96 [inlined]
typeinf_code at ./compiler/typeinfer.jl:193
typeinf_ext at ./compiler/typeinfer.jl:235
jfptr_typeinf_ext_12 at /Users/gnimuc/Codes/julia/usr/lib/julia/sys.dylib (unknown line)
jl_apply at /Users/gnimuc/Codes/julia/src/./julia.h:1538 [inlined]
jl_apply_with_saved_exception_state at /Users/gnimuc/Codes/julia/src/rtutils.c:257
jl_type_infer at /Users/gnimuc/Codes/julia/src/gf.c:274
jl_compile_method_internal at /Users/gnimuc/Codes/julia/src/gf.c:1778
jl_fptr_trampoline at /Users/gnimuc/Codes/julia/src/gf.c:1822
do_call at /Users/gnimuc/Codes/julia/src/interpreter.c:324
eval_stmt_value at /Users/gnimuc/Codes/julia/src/interpreter.c:363 [inlined]
eval_body at /Users/gnimuc/Codes/julia/src/interpreter.c:671
jl_interpret_toplevel_thunk_callback at /Users/gnimuc/Codes/julia/src/interpreter.c:788
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x11339fc6f)
unknown function (ip: 0x0)
jl_interpret_toplevel_thunk at /Users/gnimuc/Codes/julia/src/interpreter.c:797
jl_toplevel_eval_flex at /Users/gnimuc/Codes/julia/src/toplevel.c:806
jl_toplevel_eval_in at /Users/gnimuc/Codes/julia/src/builtins.c:631
eval at ./boot.jl:320
eval_user_input at /Users/gnimuc/Codes/julia/usr/share/julia/stdlib/v0.7/REPL/src/REPL.jl:85
macro expansion at /Users/gnimuc/.julia/packages/Revise/lewx/src/Revise.jl:433 [inlined]
#25 at ./task.jl:257
jl_fptr_trampoline at /Users/gnimuc/Codes/julia/src/gf.c:1823
jl_apply at /Users/gnimuc/Codes/julia/src/./julia.h:1538 [inlined]
start_task at /Users/gnimuc/Codes/julia/src/task.c:268
(a = 0, b = 1, c = nothing, d = 3)
```

**Comment by Gnimuc** [comment]
this is a problem of Julia itself, fixed in v1.0.0.

**Comment by quinnj** [comment]
Ah good! Yes, now that you mention it, I do remember seeing this error myself at one point, but then it indeed got fixed.
